sys-utils/fallocate.c

```diff
@@ -195,9 +197,10 @@ static void dig_holes(int fd, off_t file_off, off_t len)
 
 	bufsz = st.st_blksize;
 
-	if (lseek(fd, file_off, SEEK_SET) < 0)
+	if (!report && lseek(fd, file_off, SEEK_SET) < 0)
 		err(EXIT_FAILURE, _("seek on %s failed"), filename);
```

Comment:
Why is there an exception for `report`? I guess the `-o offset` can be used with --report-holes too.